docs/proposals/20230726-integrate-keda-http-add-on.md

```diff
@@ -0,0 +1,291 @@
+# Integrate KEDA http-add-on
+
+## Motivation
+OpenFunction has released v1.1.1, and [keda-http-addon](https://github.com/kedacore/http-add-on/) has been updated to v0.5.0. Now we can try to add support for keda-http-addon.
+
+In OpenFunction v1beta2 version API, `trigger` and `inputs` have been separated, so keda-http-addon can take over the part of `HttpTrigger`; on the other hand, integrating keda-http-addon also helps to unify the core architecture of OpenFunction and improve consistency. 
+
+## Associated issue
+
+- https://github.com/OpenFunction/OpenFunction/issues/162
+
+## Goals
+Integrate [keda-http-addon](https://github.com/kedacore/http-add-on/) into OpenFunction.
+
+## Proposal
+
+### Overview
+
+- **Request**:
+  1. When a function is created, a function service and an HTTPScaledObject resource are created for it under the same namespace (HTTPScaledObject will automatically create the corresponding ScaledObject resource).  
+  2. Users access the service through the address exposed by Ingress.
+  3. The requests reach the Interceptor Service through the entry point and are then passed to the Interceptor Deployment.
+  4. Interceptor Deployment delivers the requests to the corresponding Function Service based on the information in the Routing Table, which then receives and processes it.
+- **Auto-scaling**:
+  1. In step 3 of the Request process, the Interceptor Deployment counts the requests, which are periodically collected and aggregated by the External Scaler Deployment.
+  2. External Scaler Deployment reports request counts to the KEDA operator.
+  3. KEDA scales the target (Function) that reaches the threshold value according to the scaling options.
+- **Inject Dapr sidecar**:
+  1. Dapr control plane includes the [dapr-sidecar-injector service](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-overview/), which watches for new pods with the `dapr.io/enabled` annotation and injects a container with the `daprd` process within the pod.
+  2. In this case, sidecar arguments can be passed through annotations as outlined in the **Kubernetes annotations** column in [this table](https://docs.dapr.io/reference/arguments-annotations-overview/).
+
+![sync runtime overview](../images/integrate_keda_http-add-on.png)
+
+![KEDA+Dapr architecture overview](../images/function-keda-dapr.png)
+
+### CRD
+
+Add an `HTTPScaledObject ` type in `OpenFunction/apis/core/v1beta2/serving_types.go`:
+
+```go
+type HTTPScaledObject struct {
+	// Minimum amount of replicas to have in the deployment (Default 0)
+	// +optional
+	MinReplicaCount *int32 `json:"minReplicaCount,omitempty"`
```

Comment:
We had defined the `MinReplicaCount` and `MaxReplicaCount` [here](https://github.com/OpenFunction/OpenFunction/blob/main/apis/core/v1beta2/serving_types.go#L134C2-L134C2), no need to define in the `HTTPScaledObject`.

Reply:
I see, thanks!